x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/rank/textsimilarity/ChunkScorerConfig.java

```diff
@@ -52,16 +54,26 @@ public static ChunkingSettings chunkingSettingsFromMap(Map<String, Object> map)
     public ChunkScorerConfig(StreamInput in) throws IOException {
         this.size = in.readOptionalVInt();
         this.inferenceText = in.readString();
+        if (in.getTransportVersion().onOrAfter(TransportVersions.V_8_19_0)) {
```

Comment:
Remember our discussion about usability here? We should re-use the `inferenceId` specified in the retriever already. Nothing should be added to the chunk scoring config.